core/lib/src/server.rs

```diff
@@ -418,14 +430,26 @@ impl Rocket {
             let rocket = rocket.clone();
             let remote = conn.remote_addr().unwrap_or_else(|| ([0, 0, 0, 0], 0).into());
             async move {
-                Ok::<_, std::convert::Infallible>(hyper::service_fn(move |req| {
+                let service = hyper::service_fn(move |req| {
                     hyper_service_fn(rocket.clone(), remote, req)
-                }))
+                });
+                Ok::<_, std::convert::Infallible>(service)
             }
         });
 
-        // NOTE: `hyper` uses `tokio::spawn()` as the default executor.
+        #[derive(Clone)]
+        struct InCurrentSpanExecutor;
```

Comment:
Why is this here? Isn't this what `.instrument()` and `.in_current_span()` are for?

Reply:
This is added so that hyper's calls to `tokio::spawn` will "inherit" the active spans (e.g. the current connection, request, etc.), instead of being logged at the root level.